libr/socket/socket_http.c

```diff
@@ -3,35 +3,41 @@
 #include <r_socket.h>
 #include <r_util.h>
 
-static int __socket_slurp (RSocket *s, ut8 *buf, int bufsz) {
-	int i;
-	int chsz = 1;
-	// r_socket_block_time (s, 1, 1, 0);
-	if (r_socket_read_block (s, (ut8 *) buf, 1) != 1) {
+static size_t __socket_slurp(RSocket *s, RBuffer *buf) {
+	size_t i;
+	ut8 c;
+	r_socket_block_time (s, 1, 1, 0);
+	if (r_socket_read_block (s, &c, 1) != 1) {
 		return 0;
 	}
-	for (i = 1; i < bufsz; i += chsz) {
-		buf[i] =0;
+	r_buf_append_bytes (buf, &c, 1);
+	for (i = 1;; i += 1) {
```

Comment:
What if the server keep sending data? Can this result in infinite allocation of data on the client side?